### PR TITLE
EES-4622/Add regulation info to table tool info

### PR DESCRIPTION
**Pull request by sambiramairelogic**
Marking as draft just because I anticipate some change requests ;) 

This adds new copy to each occurrence of the `TableToolInfo` info, specifically to display the Release Type and hint modal. To obtain the ReleaseType in this component:

* For its occurrence in frontend I've extended the `FastTrackTableAndReleaseMeta` (TS) and `FastTrackViewModel` (CS) types to carry the ReleaseType
* For its occurrence in admin-prototypes I've extended the `SelectedPublication` (TS) type to carry the ReleaseType. This has a TODO to be removed in EES-4312, but so we might want to consider doing that now rather than extending a short-lived type? This did not require a backend change as it already existed on the view model.
* For its occurrence in admin-releasetabletool I've had to make an additional API request through the `releaseService`, as it didn't feel appropriate to extend the `publication` or `datablock` types with the ReleaseType property. 
* For its occurrence in admin-prereleasetabletool I've also had to make an additional API request for the same reasons. 

In addition, I've moved the `ReleaseTypesModal` from frontend into common so that it can be used within the `TableToolInfo` component.

I'd ideally like to avoid making the additional API requests just to get a single type, but don't see a way around it atm. 

EES-4622-changes
EES-4622-admin-prereleasetabletoolpage-changes
EES-4622-admin-prototype-changes
EES-4622-admin-releasetabletoolpage-changes

### CODE REVIEW

File: src/explore-education-statistics-common/src/modules/table-tool/types/selectedPublication.ts [hunk 1]
```diff
@@ -15,4 +17,5 @@ export interface SelectedRelease {
   slug: string;
   latestData: boolean;
   title: string;
+  type: ReleaseType;
```
Comment (sambiramairelogic) [inline src/explore-education-statistics-common/src/modules/table-tool/types/selectedPublication.ts:20]: This is the short-lived type extension I mentioned above. I'd loathe to add this here, but conscious of scope creep if we do EES-4312 in this PR. Happy to go either way, just wanna flag it :)

Comment (ntsim) [reply src/explore-education-statistics-common/src/modules/table-tool/types/selectedPublication.ts:20]: It's fine, eventually we'll clean this up :smile:

File: src/explore-education-statistics-admin/src/pages/release/content/components/ReleasePreviewTableToolFinalStep.tsx [hunk 2]
```diff
@@ -32,13 +33,15 @@ interface ReleasePreviewTableToolFinalStepProps {
   table: FullTable;
   tableHeaders: TableHeadersConfig;
   onReorderTableHeaders: (reorderedTableHeaders: TableHeadersConfig) => void;
+  releaseType: ReleaseType;
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-admin/src/pages/release/content/components/ReleasePreviewTableToolFinalStep.tsx]: Done

File: src/explore-education-statistics-admin/src/pages/release/content/components/ReleasePreviewTableToolFinalStep.tsx [hunk 3]
```diff
@@ -32,13 +33,15 @@ interface ReleasePreviewTableToolFinalStepProps {
...
 }
 const ReleasePreviewTableToolFinalStep = ({
   publication,
   query,
   table,
   tableHeaders,
   onReorderTableHeaders,
+  releaseType,
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-admin/src/pages/release/content/components/ReleasePreviewTableToolFinalStep.tsx]: Done

File: src/explore-education-statistics-admin/src/pages/release/datablocks/ReleaseTableToolPage.tsx [hunk 4]
```diff
@@ -6,19 +6,36 @@ import {
...
-  const { value: publication, isLoading } = useAsyncHandledRetry(
-    () => publicationService.getPublication(publicationId),
-    [releaseId],
-  );
+  const { value: publicationAndRelease, isLoading } = useAsyncHandledRetry<
+    PublicationAndRelease | undefined
+  >(async () => {
+    const [publication, release] = await Promise.all([
+      publicationService.getPublication(publicationId),
+      releaseService.getRelease(releaseId),
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-admin/src/pages/release/datablocks/ReleaseTableToolPage.tsx]: Fantastic, glad we can avoid the extra call! Changed :)

File: src/explore-education-statistics-admin/src/pages/release/datablocks/ReleaseTableToolPage.tsx [hunk 5]
```diff
@@ -6,19 +6,36 @@ import {
...
 const ReleaseTableToolPage = ({
   match,
 }: RouteComponentProps<ReleaseRouteParams>) => {
   const { releaseId, publicationId } = match.params;
 
-  const { value: publication, isLoading } = useAsyncHandledRetry(
-    () => publicationService.getPublication(publicationId),
-    [releaseId],
-  );
+  const { value: publicationAndRelease, isLoading } = useAsyncHandledRetry<
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-admin/src/pages/release/datablocks/ReleaseTableToolPage.tsx]: All in favour of Tanstack :) Have moved this over

File: src/explore-education-statistics-common/src/modules/table-tool/components/TableToolInfo.tsx [hunk 6]
```diff
@@ -1,27 +1,47 @@
...
       <h3>Related information</h3>
+
+      {releaseType && (
+        <>
+          <p>Release Type: {releaseTypes[releaseType]}</p>
+          <div className="govuk-hint">
+            <ReleaseTypesModal />
+          </div>
+        </>
+      )}
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-common/src/modules/table-tool/components/TableToolInfo.tsx]: Ah fair enough, the ticket just says "With help button like on find stats. linking to glossary entry" so I was just reproducing it as it appears on that page. Have updated to reflect your expectations.

File: src/explore-education-statistics-common/src/modules/table-tool/components/TableToolWizard.tsx [hunk 7]
```diff
@@ -50,6 +51,7 @@ export interface InitialTableToolState {
     table: FullTable;
     tableHeaders: TableHeadersConfig;
   };
+  releaseType?: ReleaseType;
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-common/src/modules/table-tool/components/TableToolWizard.tsx]: Indeed, removed

File: src/explore-education-statistics-frontend/src/modules/find-statistics/components/FilterModals.tsx [hunk 8]
```diff
@@ -5,57 +5,7 @@ import SummaryListItem from '@common/components/SummaryListItem';
...
-        usually based on administrative data but can also be a product of survey
-        data. The terms administrative data and management information are
-        sometimes used interchangeably.
-      </SummaryListItem>
-    </SummaryList>
-  </Modal>
-);
-
-export const ThemesModal = ({ themes }: { themes: ThemeSummary[] }) => (
+const ThemesModal = ({ themes }: { themes: ThemeSummary[] }) => (
```
[the remark this answers is not in the document]
Comment (sambiramairelogic) [reply src/explore-education-statistics-frontend/src/modules/find-statistics/components/FilterModals.tsx]: Yeah agreed, I contemplated doing this at the time but wanted to make minimal changes. Done now

**Review by ntsim — APPROVED** [review]
